src/Internal/Payment/PaymentContext.php

```diff
@@ -300,6 +307,24 @@ public function get_transitions(): array {
 		return $this->transitions;
 	}
 
+	/**
+	 * Stores state exception.
+	 *
+	 * @param \Exception $exception Exception to store.
+	 */
+	public function set_exception( $exception ) {
```

Comment:
I'm not insisting on this, but I'd prefer to call those `set_last_exception` and `get_last_exception`, similarly to how there's a `$wpdb->last_error`.

Also, please add a proper type hint to the method:
```suggestion
	public function set_exception( Exception $exception ) {
```

Similar,y `get_(last_)exception()` should return `: ?Exception`